core/src/main/java/org/dozer/propertydescriptor/PropertyDescriptorFactory.java

```diff
@@ -49,8 +50,12 @@ public static DozerPropertyDescriptor getPropertyDescriptor(Class<?> clazz, Stri
         boolean isMapProperty = MappingUtils.isSupportedMap(clazz);
         if (name.equals(DozerConstants.SELF_KEYWORD) &&
             (mapSetMethod != null || mapGetMethod != null || isMapProperty)) {
```

Comment:
I have checked one of the failing test cases. Currently Dozer does support `mapSetMethod` and `mapGetMethod` on non-`java.util.Map` objects. For instance, test uses `org.dozer.vo.map.CustomMap` which does not implement JDK `Map`, but its own interface. So apparently we cannot remove those not-null checks. What we could do, is to update documentation (quoted above) to reflect actual implementation. But to me this is out of scope of this PR.